app/models/relationship.rb

```diff
@@ -167,10 +167,17 @@ def self.create_unless_exists(source_id, target_id, relationship_type, options =
           r.send("#{key}=", value) if r.respond_to?("#{key}=")
         end
         r.save!
-      rescue
+      rescue StandardError => e
+        logger.error("[Relationship::create_unless_exists] rescuing #{e}")
+        CheckSentry.notify(e, source_id: source_id, target_id: target_id, relationship_type: relationship_type)
         r = Relationship.where(source_id: source_id, target_id: target_id).where('relationship_type = ?', relationship_type.to_yaml).last
       end
     end
+    if r.nil?
+      logger.error("[Relationship::create_unless_exists] returning nil")
```

Comment:
Opinion: This log entry, as is, is not super useful since we don't have any context information. I suggest we also include the `source_id` and `target_id`, or just drop this line, since it's going to be reported to Sentry anyway.

Minor: in models, we usually set `Rails.logger` instead of just `logger`, so I would rather keep it only for consistency.

Reply:
Done. I'd like to keep the message here to be able to follow this through in CloudWatch with the other steps already in the logs.